bundles/binding/org.openhab.binding.lightwaverf/src/main/java/org/openhab/binding/lightwaverf/internal/LightwaveRFSender.java

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.lightwaverf.internal;
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.util.concurrent.BlockingDeque;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import org.openhab.binding.lightwaverf.internal.command.LightwaveRFCommand;
+import org.openhab.binding.lightwaverf.internal.command.LightwaveRfCommandOk;
+import org.openhab.binding.lightwaverf.internal.command.LightwaveRfHeatInfoRequest;
+import org.openhab.binding.lightwaverf.internal.command.LightwaveRfRoomDeviceMessage;
+import org.openhab.binding.lightwaverf.internal.command.LightwaveRfRoomMessage;
+import org.openhab.binding.lightwaverf.internal.command.LightwaveRfSerialMessage;
+import org.openhab.binding.lightwaverf.internal.command.LightwaveRfVersionMessage;
+import org.openhab.binding.lightwaverf.internal.message.LightwaveRFMessageListener;
+import org.openhab.binding.lightwaverf.internal.message.LightwaveRfMessageId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+ /**
+  * @author Neil Renaud
+  * @since 1.7.0
+ */
+public class LightwaveRFSender implements Runnable, LightwaveRFMessageListener {
```

Comment:
since the implementations of sender and receiver contains some duplicated i would suggest to introduce a new Superclass